src/head.js

```diff
@@ -2,19 +2,23 @@ var nth = require('./nth');
 
 
 /**
- * Returns the first element in a list.
- * In some libraries this function is named `first`.
+ * Returns the first element of the given list or string. In some libraries
+ * this function is named `first`.
  *
  * @func
  * @memberOf R
  * @category List
  * @see R.tail, R.init, R.last
  * @sig [a] -> a | Undefined
- * @param {Array} list The array to consider.
- * @return {*} The first element of the list, or `undefined` if the list is empty.
+ * @sig String -> String
```

Comment:
It's necessary to include a second signature since `R.head('')` gives `''` rather than `undefined`. Should we also provide explicit signatures for functions such as `R.tail` for which the list signature *is* appropriate?

It's worth noting that this would not be a problem if the signature were `R.head :: [a] -> Maybe a`. ;)